tests/integration/test_main.py

```diff
@@ -53,13 +54,41 @@ def test_version():
 
 @patch('esmvalcore._main.ESMValTool.run', new=wrapper(ESMValTool.run))
 def test_run():
-    """Test version command"""
+    """Test version command."""
     with arguments('esmvaltool', 'run', 'recipe.yml'):
         run()
 
 
+def test_empty_run(tmp_path):
+    """Test real run with no diags."""
+    recipe_file = tmp_path / "recipe.yml"
+    content = dedent("""
+        documentation:
+          description: This is a test recipe.
+          authors:
+            - andela_bouwe
+          references:
+            - contact_authors
+            - acknow_project
+          projects:
+            - c3s-magic
+        diagnostics: null
+    """)
+    recipe_file.write_text(content)
+    Config.get_config_user(path=tmp_path)
+    with pytest.raises(check.RecipeError) as exc:
+        ESMValTool.run(recipe_file, config_file=f"{tmp_path}/config-user.yml")
+    assert str(exc.value) == 'The given recipe does not have any diagnostic.'
+    log_dir = './esmvaltool_output'
+    log_file = os.path.join(log_dir,
+                            os.listdir(log_dir)[0], 'run', 'main_log.txt')
+    os.system("rm -r esmvaltool_output")
```

Comment:
This looks very unsafe, why not use tmp_path for the output and let pytest clean up?